src/is_instance/main.py

```diff
@@ -49,21 +49,26 @@ def is_instance(obj, cls):
             key, val in obj.items()
         )
 
-    if issubclass(outer_type, (list, set, Container, Iterable, Sequence)):
+    if issubclass(outer_type, Generator):
+        raise NotImplementedError('Generator not yet supported')
+
+    if issubclass(outer_type, Iterator):
+        raise NotImplementedError('Iterator not yet supported')
+
+    if issubclass(outer_type, (Container, Iterable)):
         assert len(inner_types) == 1
         [inner_type] = inner_types
-        return all(is_instance(item, inner_type) for item in obj)
+        if len(obj):
+            return all(is_instance(item, inner_type) for item in obj)
+        return is_instance(obj, inner_type) or hasattr(obj, '__class_getitem__')
```

Comment:
This may equivalently be written as
```
# the object has a len of 0...
if hasattr(obj, '__class_getitem__'):  # ...and is subscriptable
    return True
return is_instance(obj, inner_type)  # but is not subscriptable
```